libstdc++-v3/configure.ac

```diff
@@ -212,10 +212,16 @@ GLIBCXX_CHECK_C99_TR1
 GLIBCXX_COMPUTE_STDIO_INTEGER_CONSTANTS
 
 # For gettimeofday support.
-GLIBCXX_CHECK_GETTIMEOFDAY
+AC_DEFINE(_GLIBCXX_USE_GETTIMEOFDAY)
 
 # For clock_gettime, nanosleep and sched_yield support.
-GLIBCXX_ENABLE_LIBSTDCXX_TIME
+AC_DEFINE(_GLIBCXX_USE_CLOCK_GETTIME_SYSCALL)
+AC_DEFINE(_GLIBCXX_USE_CLOCK_MONOTONIC)
+AC_DEFINE(_GLIBCXX_USE_CLOCK_REALTIME)
+AC_DEFINE(_GLIBCXX_USE_SCHED_YIELD)
+AC_DEFINE(_GLIBCXX_USE_NANOSLEEP)
+AC_DEFINE(HAVE_SLEEP)
+AC_DEFINE(HAVE_USLEEP)
```

Comment:
We should not be brute-forcing global configs like this.

[`GLIBCXX_ENABLE_LIBSTDCXX_TIME`](https://github.com/zephyrproject-rtos/gcc/blob/zephyr-gcc-12.2.0/libstdc%2B%2B-v3/acinclude.m4#L1332) itself should be modified to set Zephyr-specific configs that apply when the target is Zephyr.